Update api.md (#247)
timescale/docs.timescale.com-content: api.md

> - [attach_tablespace](#attach_tablespace)
> - [chunk_relation_size](#chunk_relation_size)
> - [chunk_relation_size_pretty](#chunk_relation_size_pretty)

> > - [Compress Table ORDER BY] (#Compress_table-orderby)

Reviewer: all lowercase for anchors
```suggestion
> - [compress table ORDER BY] (#compress_table-orderby)
```

> > -	[Compress Table SEGMENT BY] (#Compress_table-segmentby)

Reviewer: ```suggestion
> -	[compress table SEGMENT BY] (#compress_table-segmentby)
```

> > -	[Compress Chunk](#Compress_chunk)

Reviewer: ```suggestion
> -	[compress chunk](#compress_chunk)
```

> > -	[Compress Policy] (#Compress_policy)

Reviewer: ```suggestion
> -	[compress policy] (#compress_policy)
```

> > -	[Compress Policy Remove] (#Compress_policy_remove)

Reviewer: ```suggestion
> -	[compress policy remove] (#compress_policy_remove)
```

> - [create_hypertable](#create_hypertable)
> - [create index (transaction per chunk)](#create_index)
> - [create view (continuous aggregate)](#continuous_aggregate-create_view)

> > -	[Decompress Chunk] (#Decompress_Chunk)

Reviewer: ```suggestion
> -	[decompress chunk] (#decompress_chunk)
```

runs a reorder on the `_timescaledb_internal._hyper_1_10_chunk` chunk using the `conditions_device_id_time_idx` index.

---

> ## Compression :community_function: [](Compression)

Reviewer: ```suggestion
## Compression :community_function: [](compression)
```

> TimescaleDB allows users to configure thier hypertable for compression and compress chunks of data manually or based 

Reviewer: ```suggestion
TimescaleDB allows users to 
- configure their hypertables for compression 
- compress chunks of data manually or based 
```

> on policy.  Policy based compression will allow the user to compress a chunk when it has reached the specified age.

Reviewer: ```suggestion
on policy.  

Policy based compression will allow the user to compress a chunk when it has reached the specified age.
```

> The user may also manually compress a chunk or sets of chunks.  For more detailed discussion of this fuction <insert 

Reviewer: ```suggestion
The user may also manually compress a chunk or sets of chunks.  For more detailed discussion of this function <insert 
```

tag to tutorial>.

> >:WARNING: Compression is disabled when running postgres 9.6 you must upgrade your Postgresal instance to Postgresql 10

Reviewer: ```suggestion
>:WARNING: Compression is only available for Postgres 10.9+
```

 or later in order to use this feature. 
 
 #### Restrictions
 Version 1.5 does not support altering or inserting data into compressed chunks.  The data can be queried without any 
 modifications, however if you need to backfill or update data in a compressed chunk you will need to decompress the chunk(s)
 first.

*	[Compress Table ORDER BY] (#Compress_table-orderby)
*	[Compress Table SEGMENT BY] (#Compress_table-segmentby)
*	[Compress Chunk](#Compress_chunk)
*	[Compress Policy] (#Compress_policy)
*	[Compress Policy Remove] (#Compress_policy_remove)
*	[Decompress Chunk] (#Decompress_Chunk)

> ## Compress Table ORDER BY (Compression) :community_function: [](#Compress_table-orderby)

Reviewer: ```suggestion
## Compress Table ORDER BY (Compression) :community_function: [](#compress_table-orderby)
```

> ## Compress Table ORDER BY (Compression) :community_function: [](#Compress_table-orderby)

Reviewer 2: I wonder if this should really be a `ALTER TABLE (Compression)` section modeled after what we already have for "ALTER VIEW (Continuous Aggregate) " That way order by and segment_by become options within the command. This may be more intuitive and follows established conventions.

'ALTER TABLE' statement is used to set table data organization characteristics for a hypertable 

> that is setup to support compression.

Reviewer: ```suggestion
to support compression.
```

The syntax is:

``` sql
ALTER TABLE <table_name> SET (timescale.compress,timescale.compress_orderby

> = '<column_name>, <column_name>');

Reviewer: Missing a `Required Arguments` section

```
#### Sample Usage [](Compress_table-orderby)

> Set Hypertable to use ORDER BY pricipal for data organization

Reviewer 2: This is not a principal but an option.

```sql
ALTER TABLE metrics SET (timescaledb.compress, timescaledb.compress_orderby = 'device_id, time');
```

## Compress Table SEGMENT BY (Compression) :community_function: [](#Compress_table-segmentby)

'ALTER TABLE' statement is used to set table data organization characteristics for a hypertable 
that is setup to support compression.

The syntax is:

``` sql
ALTER TABLE <table_name> SET (timescaledb.compress, timescaledb.compress_orderby = 
<'column'>, timescaledb.compress_segmentby = <'column'>);
```
#### Sample Usage [](Compress_table-orderby)
Set Hypertable to use SEGMENT BY Pricipal for data organization
```sql
ALTER TABLE metrics SET (timescaledb.compress, timescaledb.compress_orderby = 
'time', timescaledb.compress_segmentby = 'device_id');

```
## Compress Chunk (Compression) :community_function: [](#Compress_chunk)

> 'SELECT' Statement is used to compression a specific chunk.

Reviewer 2: `'SELECT' Statement `-> `The compress_chunk function`

> 'SELECT' Statement is used to compression a specific chunk.

Reviewer: line not necessary

> The Syntax is:

Reviewer: ```suggestion
The syntax is:
```

``` sql

> select compress_chunk( '<chunk_name>');

Reviewer: All caps for SQL statements here and below

```
#### Sample Usage [](Compress_chunk)
In this example we will compress chunk 1_2
``` sql
select compress_chunk( '_timescaledb_internal._hyper_1_2_chunk');
```
## Compress Policy (Compression) :community_function: [](#Compress_policy)
Allows you to set a policy based on a chunks age as to when it will be compressed.

The Syntax is:

``` sql
select add_compress_chunks_policy('<table_name>', '<time interval>'::interval);
```
#### Sample Usage [](Compress_policy)
In this example we will compress chunks older than 60 days from the 'cpu' table.

``` sql
select add_compress_chunks_policy('cpu', '60d'::interval);
```
## Compress Policy Remove (Compression) :community_function: [](#Compress_policy_remove)
If you need to suspend or remove the compression policy. To start policy basd compression again
you will need to re-add the policy.

The Syntax:

``` sql

> select remove_compress_chunks_policy('<table_name>');

Reviewer 2: `select`-> `SELECT` in a bunch of places

If you need to suspend or remove the compression policy. To start policy basd compression again
you will need to re-add the policy.

The Syntax:

``` sql
SELECT remove_compress_chunks_policy('<table_name>');

```
#### Sample Usage [](Compress_policy_remove)
In this example we will remove the compression polict from the 'conditions' table:
``` sql
SELECT remove_compress_chunks_policy('conditions');

```

## Decompress Chunk (Compression) :community_function: [](#Decompress_chunk)
If you need to modify or add data to a chunk (backfill) you will need to decompress
the chunk first using this process.

>:TIP: Prior to decompressing chunks for the purpose of data backfill or updating you should
first stop any compression policy that is active on the hypertable you plan to perfrom this 
operation on.  Once the update and/or back fill is complete simplt turn the policy back on
and the system will recompress your chucks.
 
The Syntax is:

``` sql
SELECT decompress_chunk( '<chunk_name>'); 

```
#### Sample Usage [](Decompress_chunk)
In this example we are decompressing chunk 2_2

``` sql
SELECT decompress_chunk( '_timescaledb_internal._hyper_2_2_chunk');  

```
---

> ## Move Chunks :enterprise_function: [](move-chunks)

Author: done

TimescaleDb allows users to move data chunks (and indexes) to alternative tablespaces.
This allows the user the ability to move data as it ages off to more cost effective storage.

> *	[move chunk] (#move_chunk)

Author: removed

> ## Move Chunk and Index to different table space (move-chunks) :enterprise_function: [](#move-chunks)

Author: OK I have simplified this section we really dont need multiple examples when one will do the trick

 modifications, however if you need to backfill or update data in a compressed chunk you will need to decompress the chunk(s)
 first.

*	[alter table (compression)] (#compression_alter-table))
*	[compress Table SEGMENT BY] (#compress_table-segmentby)
*	[compress Chunk](#compress_chunk)
*	[compress Policy] (#compress_policy)
*	[compress Policy Remove] (#compress_policy_remove)
*	[decompress Chunk] (#decompress_chunk)

## ALTER TABLE (Compression) :community_function: [](compression_alter-table)

'ALTER TABLE' statement is used to set table data organization characteristics for a hypertable 
to support compression.

#### Parameters
|Name|Description|
|---|---|
| `table_name` | Name of the hypertable that will support compression |
| `column_name` | name of the column used to order by and/or segment by|

#### with cluase options [](compression_alter-table)
|Name|Description|
|---|---|
| `timescale.compress_orderby` | Use of order by to organize data within hypertable |
| `timescaledb.compress_segmentby` | Use of segment by to organize data within hypertable|

The syntax (ORDER BY ONLY) is:

``` sql

> ALTER TABLE <table_name> SET (timescale.compress,timescale.compress_orderby

Reviewer: Please see `Continuous aggregates` section in api.md for example on syntax (or postgreSQL docs)  Should include all options (order by, segment by) and all parameters

Reviewer: for example, shouldn't have two `<column_name` values, but should denote one with brackets to include additional ones

Author: fixed

= '<column_name>, <column_name>');
```
#### Sample Usage [](compression_alter-table)
Set Hypertable to use ORDER BY option for data organization

```sql
ALTER TABLE metrics SET (timescaledb.compress, timescaledb.compress_orderby = 'device_id, time');
```

The syntax (ORDER BY and SEGMENT BY) is:

``` sql
ALTER TABLE <table_name> SET (timescaledb.compress, timescaledb.compress_orderby = 
<'column'>, timescaledb.compress_segmentby = <'column'>);
```

> #### Sample Usage [](compression_alter-table)

Reviewer: The two sample usage sections should be combined under one header

Author: done

Set Hypertable to use ORDER BY and SEGMENT BY options for data organization
```sql
ALTER TABLE metrics SET (timescaledb.compress, timescaledb.compress_orderby = 
'time', timescaledb.compress_segmentby = 'device_id');
```

## Compress Chunk (Compression) :community_function: [](compress_chunk)

compress_chunk function is used to compress a specific chunk.

The syntax is:

``` sql
SELECT compress_chunk( '<chunk_name>');
```

> #### Sample Usage [](compress_chunk)

Reviewer: ```suggestion
#### Sample Usage [](compress_chunk-sample-usage)
```
This section should go last

In this example we will compress chunk 1_2
``` sql
SELECT compress_chunk( '_timescaledb_internal._hyper_1_2_chunk');
```

#### Required Arguments [](compress_chunk-required-arguments)

|Name|Description|
|---|---|
| `chunk_name` | Name of the chunck to be compressed|

## Compress Policy (Compression) :community_function: [](compress_policy)
Allows you to set a policy based on a chunks age as to when it will be compressed.

The Syntax is:

``` sql
SELECT add_compress_chunks_policy('<table_name>', '<time interval>'::interval);
```

> #### Sample Usage [](compress_policy)

Reviewer: ```suggestion
#### Sample Usage [](compress_policy-sample-usage)
```

In this example we will compress chunks older than 60 days from the 'cpu' table.

``` sql
SELECT add_compress_chunks_policy('cpu', '60d'::interval);
```
#### Required Arguments [](compress_policy-required-arguments)

|Name|Description|
|---|---|
| `table_name` | Name of the the policy will act on.|
| `time_interval` | Age at which the policy job will compress chunks.|

## Compress Policy Remove (Compression) :community_function: [](compress_policy_remove)
If you need to suspend or remove the compression policy. To start policy basd compression again
you will need to re-add the policy.

The Syntax:

``` sql
SELECT remove_compress_chunks_policy('<table_name>');

```

> #### Sample Usage [](compress_policy_remove)

Reviewer: ```suggestion
#### Sample Usage [](compress_policy_remove-sample-usage)
```

> In this example we will remove the compression polict from the 'conditions' table:

Reviewer: ```suggestion
In this example we will remove the compression policy from the 'conditions' table:
```

``` sql
SELECT remove_compress_chunks_policy('conditions');
```
#### Required Arguments [](compress_policy_remove-required-arguments)

|Name|Description|
|---|---|
| `table_name` | Name of the table the policy should be removed from.|

## Decompress Chunk (Compression) :community_function: [](decompress_chunk)
If you need to modify or add data to a chunk (backfill) you will need to decompress
the chunk first using this process.

>:TIP: Prior to decompressing chunks for the purpose of data backfill or updating you should

> first stop any compression policy that is active on the hypertable you plan to perfrom this 

Reviewer: ```suggestion
first stop any compression policy that is active on the hypertable you plan to perform this 
```

> operation on.  Once the update and/or back fill is complete simplt turn the policy back on

Reviewer: ```suggestion
operation on.  Once the update and/or backfill is complete simply turn the policy back on
```

and the system will recompress your chucks.
 
The Syntax is:

``` sql
SELECT decompress_chunk( '<chunk_name>'); 

```
#### Sample Usage [](decompress_chunk)
In this example we are decompressing chunk 2_2

``` sql
SELECT decompress_chunk( '_timescaledb_internal._hyper_2_2_chunk');  
```
#### Required Arguments [](decompress_chunk-required-arguments)
|Name|Description|
|---|---|
| `chunk_name` | Name of the chunk to be decompressed.|
---

> ## Move Chunks :enterprise_function: [](move_chunk)

Reviewer: Whichever it is "move_chunk" or "move_chunks", the inconsistency needs to be resolved in the header, function calls and anchors

Author: done

> TimescaleDb allows users to move data chunks (and indexes) to alternative tablespaces.

Reviewer: ```suggestion
TimescaleDB allows users to move data chunks (and indexes) to alternative tablespaces.
```

> TimescaleDb allows users to move data chunks (and indexes) to alternative tablespaces.

Reviewer: ```suggestion
TimescaleDb allows users to move chunks (and indexes) to alternative tablespaces.
```

Author: done

> This allows the user the ability to move data as it ages off to more cost effective storage.

Reviewer: ```suggestion
This allows the user the ability to move data as it ages to more cost effective storage.
```

> ## Move Chunk and Index to different table space (move_chunk) :enterprise_function: [](move-chunk)

Reviewer: A separate header is not required, just the arguments

Author: consolidated this section I dont think we need this many examples

The Syntax is:

``` sql
SELECT move_chunk(chunk=>'<chunk_name>', destination_tablespace=>'<tablespace_name>', 
index_destination_tablespace=>'<tblespace_name', reorder_index=>'chunk_index_name', 
verbose=>TRUE);  

```

> #### Sample Usage [](move_chunk)

Reviewer: ```suggestion
#### Sample Usage [](move_chunk-sample-usage)
```

Move both the chunk and index to the same tablespace:

``` sql
SELECT move_chunk(chunk=>'_timescaledb_internal._hyper_1_4_chunk', destination_tablespace=>
'tablespace_2', index_destination_tablespace=>'tablespace_2', reorder_index=>
'_timescaledb_internal._hyper_1_4_chunk_netdata_time_idx', verbose=>TRUE);
  
```
Move the same set of chunks back to the default tablespace:

``` sql
SELECT move_chunk(chunk=>'_timescaledb_internal._hyper_1_4_chunk', destination_tablespace=>'pg_default',
index_destination_tablespace=>'pg_default', reorder_index=>
'_timescaledb_internal._hyper_1_4_chunk_netdata_time_idx', verbose=>TRUE);

```
Move Chunk to one tablespace and the index to another:

``` sql
SELECT move_chunk(chunk=>'_timescaledb_internal._hyper_1_4_chunk', destination_tablespace=>'tablespace2',
index_destination_tablespace=>'tablespae3', reorder_index=>
'_timescaledb_internal._hyper_1_4_chunk_netdata_time_idx', verbose=>TRUE);

```

> #### Required Arguments [](compress_policy-required-arguments)

Reviewer: Needs to come right after syntax section

Reviewer: wrong anchor name
```suggestion
#### Required Arguments [](move_chunk-required-arguments)
```
